full-service/src/db/transaction_log.rs

```diff
@@ -2123,4 +2160,226 @@ mod tests {
+            .unwrap();
+        builder.set_tombstone(0).unwrap();
+        builder.select_txos(conn, None).unwrap();
+        let unsigned_tx_proposal = builder
+            .build(
+                TransactionMemo::RTH {
+                    subaddress_index: None,
+                },
+                conn,
+            )
+            .unwrap();
+        let tx_proposal = unsigned_tx_proposal.clone().sign(&account).await.unwrap();
+
+        // Log submitted transaction from tx_proposal
+        let first_account_second_tx_log = TransactionLog::log_submitted(
+            &tx_proposal,
+            ledger_db.num_blocks().unwrap(),
+            "".to_string(),
+            &first_account_id.to_string(),
+            conn,
+        )
+        .unwrap();
+
+        let key_images: Vec<KeyImage> = tx_proposal
+            .input_txos
+            .iter()
+            .map(|txo| txo.key_image)
+            .collect();
+
+        // Note: This block doesn't contain the fee output.
+        add_block_with_tx_outs(
+            &mut ledger_db,
+            &[
+                tx_proposal.change_txos[0].tx_out.clone(),
+                tx_proposal.payload_txos[0].tx_out.clone(),
+            ],
+            &key_images,
+            &mut rng,
+        );
+
+        assert_eq!(ledger_db.num_blocks().unwrap(), 17);
+
+        // build a transaction from second_account
+        let account = Account::get(&second_account_id, conn).unwrap();
+
+        let (recipient, mut builder) =
+            builder_for_random_recipient(&second_account_key, &ledger_db, &mut rng);
+        builder
+            .add_recipient(recipient.clone(), 55 * MOB, Mob::ID)
+            .unwrap();
+        builder.set_tombstone(0).unwrap();
+        builder.select_txos(conn, None).unwrap();
+        let unsigned_tx_proposal = builder
+            .build(
+                TransactionMemo::RTH {
+                    subaddress_index: None,
+                },
+                conn,
+            )
+            .unwrap();
+        let tx_proposal = unsigned_tx_proposal.clone().sign(&account).await.unwrap();
+
+        // Log submitted transaction from tx_proposal
+        let second_account_tx_log = TransactionLog::log_submitted(
+            &tx_proposal,
+            ledger_db.num_blocks().unwrap(),
+            "".to_string(),
+            &second_account_id.to_string(),
+            conn,
+        )
+        .unwrap();
+
+        let key_images: Vec<KeyImage> = tx_proposal
+            .input_txos
+            .iter()
+            .map(|txo| txo.key_image)
+            .collect();
+
+        // Note: This block doesn't contain the fee output.
+        add_block_with_tx_outs(
+            &mut ledger_db,
+            &[
+                tx_proposal.change_txos[0].tx_out.clone(),
+                tx_proposal.payload_txos[0].tx_out.clone(),
+            ],
+            &key_images,
+            &mut rng,
+        );
+
+        assert_eq!(ledger_db.num_blocks().unwrap(), 18);
```

Comment:
I think this can be DRYed up some
```suggestion
        let recipient_account_key = AccountKey::random(&mut rng);
        let recipient = recipient_account_key.subaddress(rng.next_u64());
        let mut logs = vec![];
        for (account_key, value) in [(&account_keys[0], 50), (&account_keys[0], 45), (&account_keys[1], 55)] {
            let (log, tx_proposal) = create_test_minted_and_change_txos(account_key.clone(), recipient.clone(), value * MOB, wallet_db.clone(), ledger_db.clone()).await;
            let key_images: Vec<KeyImage> = tx_proposal
                .input_txos
                .iter()
                .map(|txo| txo.key_image)
                .collect();

            // Note: This block doesn't contain the fee output.
            add_block_with_tx_outs(
                &mut ledger_db,
                &[
                    tx_proposal.change_txos[0].tx_out.clone(),
                    tx_proposal.payload_txos[0].tx_out.clone(),
                ],
                &key_images,
                &mut rng,
            );
            logs.push(log);
        }
        assert_eq!(ledger_db.num_blocks().unwrap(), 18);
```

Reply:
removed this code from the PR